docs/architecture/adr-100-data-companion-push-api.md

```diff
@@ -0,0 +1,441 @@
+   IBC relayer stability (see [\#6729] and [\#7156]).
+2. **Block execution result storage**, which is required to facilitate a number
+   of RPC APIs, but also results in storage of large quantities of data that is
+   not critical to consensus. This data is, however, critical for certain use
+   cases outside of consensus.
+
+Another intersecting issue is that there may be a variety of use cases that
+require different data models. A good example of this is the mere existence of
+the [PostgreSQL indexer] as compared to the default key/value indexer.
+
+Continuing from and expanding on the ideas outlined in [RFC 006][rfc-006], the
+suggested approach in this ADR is to provide a mechanism to publish certain data
+from Tendermint, in real-time and with certain reliability guarantees, to a
+single companion service outside of the node that can use this data in whatever
+way it chooses (filter and republish it, store it, manipulate or enrich it,
+etc.).
+
+Specifically, this mechanism would initially publish:
+
+1. Block data (i.e. `Block` data structures that have been committed by the
+   consensus engine).
+2. The results of block execution (e.g. data from `FinalizeBlockResponse`). This
+   data is not accessible from the P2P layer, and currently provides valuable
+   information for Tendermint integrators (whether events should be handled at
+   all by Tendermint is a different problem).
+
+## Alternative Approaches
+
+1. One clear alternative to this would be the approach outlined in [ADR
+   075][adr-075]. This approach:
+
+   1. Still leaves Tendermint responsible for maintaining a query interface and
+      event indexing functionality, increasing the long-term maintenance burden of,
+      and the possibility of feature sprawl in, that subsystem. To overcome this,
+      we could remove the query interface altogether and just always publish all
+      data.
+   2. Only keeps track of a sliding window of events (and does so in-memory), and
+      so does not provide reliable guaranteed delivery of this data. Even just
+      persisting this data will not solve this problem: if we prioritize
+      continued operation of the node over guaranteed delivery of this data, we
+      may eventually run out of disk storage. So here we would need to persist the
+      data _and_ we would need to panic if the sliding window of events fills up.
+   3. Allows for multiple consumers. While desirable from typical web services,
+      Tendermint is a consensus engine and, as such, we should prioritize consensus
+      over providing similar guarantees to a scalable RPC interface (especially
+      since we know this does not scale, and if such a job could just be
+      outsourced). If we have a single consumer, we can easily automatically prune
+      data that was already sent. We could overcome this by limiting the API to a
+      single consumer and single event window.
+
+   Making the requisite modifications to ADR-075 basically converges on the
```

Comment:
I think it would be worth saying (probably later on, when we describe this ADR's solution) that ADR075 is somewhat orthogonal to what is proposed in this ADR, in the sense that a "candidate" companion could be (a slightly modified version of) what we have today as RPC, with ADR075 on top.

Reply:
I see this is mentioned towards the end, but maybe a (forward) reference here would help the reader understand the potential "orthogonality" of the two solutions (ADR075 and ADR100)